x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_data_views.ts

```diff
@@ -9,15 +9,31 @@ import useAsync from 'react-use/lib/useAsync';
 import createContainer from 'constate';
 import { i18n } from '@kbn/i18n';
 import { findInventoryModel } from '@kbn/metrics-data-access-plugin/common';
+import { DEFAULT_METRICS_VIEW_ATTRIBUTES } from '../../../../common/constants';
+import { resolveAdHocDataView } from '../../../utils/data_view';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
 import { useLogViewReference } from '../../../hooks/use_log_view_reference';
-import { useDataView } from '../../../hooks/use_data_view';
 import { useAssetDetailsRenderPropsContext } from './use_asset_details_render_props';
 
-const useDataViews = ({ metricAlias }: { metricAlias: string }) => {
+const useDataViews = ({ metricsIndexPattern }: { metricsIndexPattern: string }) => {
   const { asset } = useAssetDetailsRenderPropsContext();
-  const { dataView: metricsDataView, loading: metricsDataViewLoading } = useDataView({
-    index: metricAlias,
-  });
+  const {
+    services: { dataViews },
+  } = useKibanaContextForPlugin();
+
+  const { value: metricsView, loading: metricsDataViewLoading } = useAsync(
```

Comment:
Are we already doing this in `useMetricsDataView`?  Would it make sense to use the `MetricsDataViewProvider` and this could be renamed to only handle the Logs data view?  I think it'd be a bit easier to understand and less code.  `AssetDetails` would use both providers.

Reply:
In fact, this hook could just handle the logs data view if we consider that it's only used in the infra plugin and is already wrapped in the `SourceContext` and `MetricsDataViewContext`.
 
But if we consider that the flyout could be used in other plugins in the future, I didn't want to create a dependency with `SourceContext`. I'll see what can be done to improve this.

Reply:
Decide to let everything integrated with `metrics_app` context providers. We'll think about how to separate the asset details from it when the time to use it in other plugin comes. Thanks for the suggestion.